### ISSUE TITLE
Terraform destroy crashing using vsphere

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-vsphere#65]
_This issue was originally opened by @davey-dev as hashicorp/terraform#10365. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

My workflow is terraform apply, run ansible playbooks and then terraform destroy.

My destroy only crashes if I run Ansible on the vsphere hosts.
```
+ terraform destroy -force -var vsphere_user=**** -var vsphere_password=****
vsphere_virtual_machine.rabbit: Refreshing state... (ID: TF-RABBIT)
vsphere_virtual_machine.cape: Refreshing state... (ID: TF-CAPE)
vsphere_virtual_machine.mongo: Refreshing state... (ID: TF-MONGO)
vsphere_virtual_machine.soc: Refreshing state... (ID: TF-SOC)
vsphere_virtual_machine.kafka: Refreshing state... (ID: TF-KAFKA)
vsphere_virtual_machine.api: Refreshing state... (ID: TF-API)
Error refreshing state: 1 error(s) occurred:

* vsphere_virtual_machine.api: unexpected EOF
panic: runtime error: index out of range
2016/11/25 10:50:13 [DEBUG] plugin: terraform: 
2016/11/25 10:50:13 [DEBUG] plugin: terraform: goroutine 29 [running]:
2016/11/25 10:50:13 [DEBUG] plugin: terraform: panic(0x2d49a80, 0xc4200120d0)
2016/11/25 10:50:13 [DEBUG] plugin: terraform: 	/opt/go/src/runtime/panic.go:500 +0x1a1
2016/11/25 10:50:13 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/vsphere.resourceVSphereVirtualMachineRead(0xc4204a0660, 0x32e6a60, 0xc42046c100, 0x1, 0x17)
2016/11/25 10:50:13 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/vsphere/resource_vsphere_virtual_machine.go:1066 +0x3458
2016/11/25 10:50:13 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc42038e7e0, 0xc4201b8960, 0x32e6a60, 0xc42046c100, 0xc4203ab1b0, 0x1, 0x18)
2016/11/25 10:50:13 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:259 +0x131
2016/11/25 10:50:13 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc420408060, 0xc4201b8910, 0xc4201b8960, 0xc4203b0ea0, 0x419de6, 
```0xc420045a68)

**Comment by javefang** [comment]
I noticed the same error when my vsphere provider was updated from 0.2.2 to 0.4.2 one day. Reverting to 0.2.2 fixed the issue.

Had some play with the code. It looks like the "index out of range" error happens when terraform tries to read a non-exist network interface by index. (see reference to code below).

https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/resource_vsphere_virtual_machine.go#L1243

Specifically

`key := mvm.Guest.Net[nidx].DeviceConfigId`

On the node that works, "mvm.Guest.Net" usually contains multiple interfaces. But on some node, the variable only contains a single interface. But the "nidx" is still 2 or something else, which caused the "index out of range" error.

**Comment by vancluever** [comment]
Hey @javefang do you have a config that reproduces this issue? Thanks!

**Comment by javefang** [comment]
I ends up fully destroy the infrastructure with 0.2.2, upgrade provider to 0.4.2 and recreate the infrastructure. The problem went away since then. So I guess it's fine for now. Thanks!

**Comment by vancluever** [comment]
Glad it worked out @javefang!

I think this one is fixed so I'm going to close it. We have done a lot of work in trying to clean up interface discovery over the last few months and with the recent regression fix in v0.4.2 at the very least this issue is not necessarily relevant contextually to any currently present bugs.

If anyone is still having issues, feel free to open a new issue with more detail!
